src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -89,7 +91,7 @@ public class BlockStoreQueue : IBlockStore, IDisposable
             this.nodeLifetime = nodeLifetime;
             this.storeSettings = storeSettings;
             this.chain = chain;
-            this.blockRepository = blockRepository;
+            this.blockRepository = new BlockRepository(network, dataFolder, dateTimeProvider, loggerFactory);
```

Comment:
Hmm I think you should still inject this and not initialize it here.
This component still owns the class but we let DI manage the lifetime.
Its also useful for mocking tests.

Reply:
Ok, I don't mind. 
Will do it in the first PR for a separated branch

Reply:
Yeah I would even do the same for the tree in CM (CM own is but lifetime is from the DI)

Reply:
I actually think this is better way than the DI